### PR TITLE
fix: add miner stats pagination by block range

### LINKED ISSUE #223: Paginated Charts via block ranges
**Describe the feature**
Similar to how the transactions list on the address page works, the charts on dashboard and `/stats/miners` should be able to be paginated by a block range instead of just `last 100 blocks`

**Additional Information**

image

### PR DESCRIPTION
**Pull request by shanejonas**
image

fixes #223

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -88,6 +88,10 @@ function App(props: any) {
   useEffect(() => {
     if (selectedNetwork && selectedNetwork.name !== query.network) {
       setQuery({ network: selectedNetwork.name });
+      history.push({
+        pathname: history.location.pathname,
+        search: `?network=${selectedNetwork.name}`,
```
Comment (meowsbits) [inline src/App.tsx]: Will this work if there are other parameters in the search pattern of the URI?

File: src/components/BlockList/BlockList.tsx [hunk 2]
```diff
@@ -60,29 +60,41 @@ function BlockList({ blocks }: any) {
...
+                  <Typography>
+                    <Link
+                      component={({ className, children }: { children: any, className: string }) => (
+                        <RouterLink className={className} to={`/address/${b.miner}`} >
+                          {children}
+                        </RouterLink>
+                      )}>
+                      {authorHashShort}
+                    </Link>
+                    &nbsp;<sup>{hexToString(b.extraData).substring(0, 20)}</sup>
```
Comment (meowsbits) [inline src/components/BlockList/BlockList.tsx]: Eventually we might want to refactor this address/+extradata format out to a configurable/standard, eg. substring length, abbreviations, sanitization, ...

File: src/components/StatCharts/StatCharts.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  return {
+    x: hexToNumber(block.number),
+    y: block.uncles.length,
+  };
+};
+
+const blockMapHashRate = (block: any) => {
+  return {
+    x: hexToNumber(block.number),
+    y: hashesToGH(new BigNumber(block.difficulty, 16).dividedBy(config.blockTime)),
```
Comment (meowsbits) [inline src/components/StatCharts/StatCharts.tsx]: Mmmm.. Using a hardcoded block time is a liiiitleee stretchy?

File: src/App.tsx (hunk 1 above)
Comment (shanejonas) [reply src/App.tsx]: it does for certain keys, see here: https://github.com/etclabscore/expedition/blob/master/src/App.tsx#L38

File: src/components/BlockList/BlockList.tsx (hunk 2 above)
Comment (shanejonas) [reply src/components/BlockList/BlockList.tsx]: agreed. lets make a ticket for that

File: src/components/StatCharts/StatCharts.tsx (hunk 3 above)
Comment (shanejonas) [reply src/components/StatCharts/StatCharts.tsx]: yea would be nice to calculate the block time for the list of blocks you are looking at. maybe a helper `getBlockTimeFromBlocks(blocks)`

File: src/App.tsx (hunk 1 above)
Comment (meowsbits) [reply src/App.tsx]: Gotcha, thanks. Maybe the supported params should be parts of a config object and standardized/globalized then too, toward fewer magic strings.

Would suggest new ticket for it.

File: src/components/BlockList/BlockList.tsx (hunk 2 above)
Comment (meowsbits) [reply src/components/BlockList/BlockList.tsx]: https://github.com/etclabscore/expedition/issues/234

File: src/App.tsx (hunk 1 above)
Comment (meowsbits) [reply src/App.tsx]: ... Or modify the handrolled `History` logic to use either a library (there _must be one_...?) and/or/thereby support arbitrary URI path/parameter/hash updates.

**Pull request merged**